Assets/HoloToolkit/Input/Scripts/Utilities/Interactions/TwoHandManipulatable.cs

```diff
@@ -0,0 +1,454 @@
+
+        /// <summary>
+        /// Maps input id -> position of hand
+        /// </summary>
+        private readonly Dictionary<uint, Vector3> m_handsPressedLocationsMap = new Dictionary<uint, Vector3>();
+
+        /// <summary>
+        /// Maps input id -> input source. Then obtain position of input source using currentInputSource.TryGetGripPosition(currentInputSourceId, out inputPosition);
+        /// </summary>
+        private readonly Dictionary<uint, IInputSource> m_handsPressedInputSourceMap = new Dictionary<uint, IInputSource>();
+
+        /// <summary>
+        /// Property that turns on and off the Visibility of the BoundingBox cloned from the BoundingBoxPrefab reference.
+        /// </summary>
+        private bool ShowBoundingBox
+        {
+            set
+            {
+                if (boundingBoxPrefab != null)
+                {
+                    if (boundingBoxInstance == null)
+                    {
+                        // Instantiate Bounding Box from the Prefab
+                        boundingBoxInstance = Instantiate(boundingBoxPrefab) as BoundingBox;
+                    }
+
+                    if (value)
+                    {
+                        boundingBoxInstance.Target = this.gameObject;
+                        boundingBoxInstance.gameObject.SetActive(true);
+                    }
+                    else
+                    {
+                        boundingBoxInstance.Target = null;
+                        boundingBoxInstance.gameObject.SetActive(false);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// SetManipulationMode
+        /// </summary>
+        public void SetManipulationMode(TwoHandedManipulation mode)
+        {
+            ManipulationMode = mode;
+        }
+
+        /// <summary>
+        /// Private Methods
+        /// </summary>
+        private void Awake()
+        {
+            m_moveLogic = new TwoHandMoveLogic();
+            m_rotateLogic = new TwoHandRotateLogic(ConstraintOnRotation);
+            m_scaleLogic = new TwoHandScaleLogic();
+        }
+
+        private void Start()
+        {
+            if (HostTransform == null)
+            {
+                HostTransform = transform;
+            }
+        }
+
+        private void Update()
+        {
+            //Update positions of all hands
+            foreach (var key in m_handsPressedInputSourceMap.Keys)
+            {
+                var inputSource = m_handsPressedInputSourceMap[key];
+                Vector3 inputPosition = Vector3.zero;
+                if (inputSource.TryGetGripPosition(key, out inputPosition))
+                {
+                    m_handsPressedLocationsMap[key] = inputPosition;
+                }
+            }
+
+            if (currentState != State.Start)
+            {
+                UpdateStateMachine();
+            }
+        }
+
+        private Vector3 GetInputPosition(InputEventData eventData)
```

Comment:
Seems like this method could just be added to the `InputSource` itself.
But `InputSource.TryGetGripPosition` does essentially the same thing. I'll address this in the input refactor.